### PR TITLE
complex: add commonly used complex number funcs

### PR DESCRIPTION
**Pull request by gerard5**
Provide required `complex.h` function prototypes and commonly used functions.

`cabs()` and `cabsf()` implementations, to calculate distance, temporarily uses `sqrt()` instead of `hypot()` which is required but missing now.

In `cexp()` implementation Euler equation is used.

In reference to https://github.com/phoenix-rtos/phoenix-rtos-project/issues/785
JIRA: RTOS-528

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: `imxrt1176-nil`, `ia32-generic`.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: include/complex.h [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+static inline float complex conjf(float complex z)
+{
+	union {
+		float complex z;
+		struct {
+			float re;
+			float im;
+		} part;
+	} u = { .z = z };
```
Comment (agkaminski) [inline include/complex.h:76]: Uhh, I don't really like this union, specifically assuming _Complex implementation, on which we have no control.

File: include/complex.h [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+{
+	union {
+		double complex z;
+		struct {
+			double re;
+			double im;
+		} part;
+	} u = { .z = z };
+	u.part.im = -u.part.im;
+	return u.z;
```
Comment (agkaminski) [inline include/complex.h:64]: I'd rather avoid doing that stuff if not necessary

btw, I guess it's ok to do:
`# define CMPLX(x, y) __builtin_complex ((double) (x), (double) (y))`

```suggestion
	return CMPLX(creal(z), -cimag(z));
```

Comment (gerard5) [reply include/complex.h:64]: What about this?
```C
float complex conjf2(float complex z)
{
    __imag__(z) = -(__imag__(z));
    return z;
}
```

Comment (gerard5) [reply include/complex.h]: It looks like when `-O0` with `__builtin_complex()` is used the machine code is shorter (one-two instructions).

File: math/complex.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * %LICENSE%
+ */
+
+#include <math.h>
+#include <complex.h>
+
+
+double cabs(double complex z)
+{
+	/* FIXME: due to missing hypot() implementation, temporarily sqrt() is used */
```
Comment (agkaminski) [inline math/complex.c:22]: Fun fact - `hypot()` can be implemented via `cabs()` - which should be fundamental then? :)

I guess `hypot()` should be, to not require complex numbers if not needed

**Review by agkaminski — APPROVED** [review]
LGTM

File: math/complex.c (hunk 3 above)
Comment (gerard5) [reply math/complex.c:22]: More than that, however, the `hypot()` function calculate $sqrt(x^2+y^2)$ in such a way that underflow will not occur, and overflow will only occur if the end result deserves it.

**Pull request merged**